### ISSUE TITLE
Can't enable plugin

### ISSUE THREAD

**Issue by agoldenlife** [TheDuckCow/MCprep#140]
Add another person to this issue. Running blender 2.83.3. A couple things to note:

The video says go to "File" then "User Preferences" This does not exist in the windows installation for 2.83.3. Instead you have to go to "Edit" then "Preferences"
When you attempt to install, I do not have a button that says "Install
from file". I only have a button that states "Install Add On". Mine is lower right, your was top right as well.
The installer says it installs successfully, but the add on remains grayed out. Let me know if I can do anything to help troubleshoot further. I have uninstalled and installed the addon several times.

Attached you will find a video (.mp4) in the zip file that I recorded of my install and results. Hopefully that is helpful.

I opened a new ticket, because the other person closed their because, "it just works now".  They didn't post why or how, I am assuming they may not now.  I have ran this as an administrator login as well, and I still could not get it to work.  Behaved the same exact way.

**Comment by agoldenlife** [comment]
I uninstalled 2.83.2 (wrote the wrong one above).  Installed 2.82a and tried again.  Same menu structure, and same a problem.  I click to show the "log" after it was installed and this is what I saw.

bpy.data.window_managers["WinMan"].addon_search = "mc"
bpy.context.space_data.system_bookmarks_active = 1
Modules Installed (MCprep_addon) from 'C:\\Users\\philip\\Downloads\\MCprep_addon_v3.2.3.1.zip' into 'C:\\Users\\philip\\AppData\\Roaming\\Blender Foundation\\Blender\\2.82\\scripts\\addons'
Modules Installed (MCprep_addon) from 'C:\\Users\\philip\\Downloads\\MCprep_addon_v3.2.3.1.zip' into 'C:\\Users\\philip\\AppData\\Roaming\\Blender Foundation\\Blender\\2.82\\scripts\\addons'

**Comment by agoldenlife** [comment]
It appears that it does not have anything to do with the version.  I tried 2.81 and 2.80 as well.  All of them had the same issue.  I was installing 64bit, but I gave 32bit a try and that did not seem to make any difference either.  Everytime, I am not allowed to enable the mcprep plugin.

**Comment by HillustratedMike36** [comment]
I just saw your video and I noticed you did not click the box--that enables the addon.
![Welp](https://user-images.githubusercontent.com/12376365/87938590-cfcbc600-ca64-11ea-978a-137c0952c494.png)

When you do that, you should get a tab labeled 'MCprep' below the 'View' tab.
![MCPrepTabWelp](https://user-images.githubusercontent.com/12376365/87939007-5f717480-ca65-11ea-881a-05882738c563.png)

**Comment by TheDuckCow** [comment]
Thank you @HillustratedMike36 for beating me to the reply! Indeed, that is likely the issue - let me know @agoldenlife if you're still having troubles.

**Comment by agoldenlife** [comment]
Thanks for taking a look and rresponding.  Yea, you could not tell in the video, but I was clicking it.  It doesn't change at all.  I have tried several different ways and accounts but I can't check the box.  It appears to install properly; I just can't enable it after installing.  Several reboots, etc... have all been tried.
________________________________

**Comment by HillustratedMike36** [comment]
_...Okay, well, that stinks._

Going back to your initial post (and the reply on the other ticket), you said you were running Blender version **2.83.3**. How...? This is my main suspicion. As of this post on the Blender website, the latest version on the official branch is **2.83.2**, which is where my screenshots came from.

Another thing I am going to ask is other addons you have. Do you have any other community addons? Maybe they are conflicting with MCprep.

**Comment by agoldenlife** [comment]
Yea, sorry.  I did not say that correctly, it was 2.83.2 not .3.  I was typing to fast and didn't proof read it.

This is the only addon we are using.  My son is just starting out and wanting to create minecraft videos.  I didn't have any of this before. :)

I don't know if it would be useful, but I could use a sysinternal tools (procmon) to see everything the install does.  I could also watch while I attempt to "enable" it and see if blender throws something up.  

I am also going to try it on a second machine to see if I get different results as well.

And I will do some searching to see what Blender has to say when someone can't enable an addon.

**Comment by TheDuckCow** [comment]
Firstly, I'd try one more thing - try running the following command in the blender interactive python console (if you are at the state of: MCprep checkbox is there, but can't enable):
```
bpy.ops.preferences.addon_enable(module="MCprep_addon")
```
![install_mcprep_cmd](https://user-images.githubusercontent.com/2958461/88113821-9ab49600-cb80-11ea-8c71-bc0f2ad78162.png)

If this doesn't work, go through the steps below instead to grab console output (which is different from the logs you find in the interface):

1) Completely remove the MCprep addon by clicking the remove icon
2) Close blender
3) Reopen blender normally, then under window menu at the top, select "toggle system console"
4) Take note of what is currently in the console output (ie, what to ignore)
5) Install blender from the .zip exactly as shown in this more recently [updated tutorial](https://www.youtube.com/watch?v=9hCkA_J019Q) (the one you mentioned sounded like it was for Blender 2.7x)
6) Note what the log output looks like in the console, copying or screenshot'ing and adding here.

Ordinarily, if there is an issue with installing the addon or an issue at the moment of enabling, it would pop up as an error. I should note, that after searching for MCprep, the addon will indeed look "grayed out" _until_ you tick the box to enable it. And again, it should either work and have a ticked box, or pop up an error message. If neither of those things are happening, and you don't have anything else printing out to the console (or info logs in the blender UI), let me know - would be open to even live chatting.

Please let me know! And sorry you're having these troubles.

**Comment by agoldenlife** [comment]
Thanks for all the help.  It is a pretty strange issue.  I did install blender on my other machine and the mcprep addon without any issues.  So it is specific to this machine, whatever the issue is.

I attempted to use the CLI as you instructed to install but got the error "addon not found".

I then followed the steps to uninstall, closed Blender, installed it again with the system console open.  I also attempted to enable it a few times as the console still shows.  Here is the results from the system console:
```
addon_utils.disable: MCprep_addon not disabled
Modules Installed (MCprep_addon) from ’C:\\Users\\philip\\Downloads\\MCprep_addon_v3.2.3.1.zip’ into 'C:\\Users\\philip\\AppData\\Roaming\\Blender Foundation\\Blender\\2.83\\scripts\\addons'
addon not found: 'MCprep_addon'
addon not found: 'MCprep_addon'
addon not found: 'MCprep_addon'
addon not found: 'MCprep_addon'
addon not found: 'MCprep_addon'
addon not found: 'MCprep_addon'

```
I also used a tools called "Procmon" to monitor the blender process and what it was attempting to do to the system to see if that gave any more clarity.   I have attached the PML files for you to view.  You will need Procmon to be able to view them.  I can also upload them using CSV if you need that as well.
[BlenderTroubleshootingLogfiles.zip](https://github.com/TheDuckCow/MCprep/files/4975670/BlenderTroubleshootingLogfiles.zip)

I couldn't come away with anything much from the procmon (process monitor) tool except that maybe it is not editing some config file back in the main directory properly.  

One other thing of note: in the video you show all the options for the add on.  I don't see that either, but I didn't know if that was because it is not enabled.

Thanks again for all the help.

**Comment by TheDuckCow** [comment]
Thanks for sharing that and the procmon outputs. Based on the output of the first step ("MCprep addon not found"), this tells me the installed zip didn't land as expect, since it can't find the module. 

I wasn't able to view the logfiles you shared (appears as a binary for me, on an OSX machine), but knowing the above let's try this step first: Manually install the .zip file.

To do this, go to the blender 2.83 addons folder, found under scripts in the User data. As [per here](https://docs.blender.org/manual/en/latest/advanced/blender_directory_layout.html#windows), would be a path like:
`%USERPROFILE%\AppData\Roaming\Blender Foundation\Blender\2.83\scripts\addons`

Now, you want to unzip the MCprep download as-is into this addons folder. You should end up with a folder named `MCprep_addon` (with none of the version numbers suffix, rename to `MCprep_addon` if different), which contains an `__init__.py` file (among other files and subfolders). If you happen to see any other rogue MCprep folders in this directory, go ahead and delete them. See the sample structure in my system below:

![mcprep_structure](https://user-images.githubusercontent.com/2958461/88751805-64e65300-d126-11ea-846c-bc56dc8702bd.png)

Now, when you open blender, you should be able to search for MCprep and enable. Let me know if this works, crossing fingers!

**Comment by agoldenlife** [comment]
We got it to work! Not quite like you mentioned, but it got me there.  It seems in Windows some files might not get installed or copied during the install process.

So, I followed your instructions.
1. Deleted the current MCPrep_addon directory.  The path you had is correct for Windows 10.
2. Unzipped the file into that directory and renamed it to MCPrep_addon
3. Started Blender, but MCPrep was not listed.
4. I then used blender to install the addon again using the zip file.  This overwrote the directory I made and I then had the same issue.
5. I then unzipped the MCPrep file and copied the contents over the existing MCPrep addon directory
**Interesting note:** It says it was copying 2006 files, but only wanted to overwrite 1911 files (which I told it to skip and not to overwrite).

I was then able to enable the add on!

Another note: Your directions might have worked.  I noticed after the fact and documenting this that my original zip extraction I made the folder too deep.  That is to say it was ...\scripts\addons\MCPrep_addon\MCPrep_addon.  Not sure how I did that, but I am guessing it would have worked if I did it your way as well.
